src/providers/ipa/ipa_init.c

```diff
@@ -259,6 +264,11 @@ static errno_t ipa_init_server_mode(struct be_ctx *be_ctx,
     sites_enabled = dp_opt_get_bool(ipa_options->basic, IPA_ENABLE_DNS_SITES);
     dnsdomain = dp_opt_get_string(be_ctx->be_res->opts, DP_RES_OPT_DNS_DOMAIN);
 
+    if (!ipa_check_fqdn(hostname)) {
```

Comment:
I'm sorry to ask for changes again, but only when I tested the patch on an IPA client, I noticed you added the check to `ipa_init_server_mode` only. That will only be called on IPA masters, but we want the check to be called every time. Can you move it to `ipa_init_misc` please?